CRM/EventsExtras/Hook/BuildForm/EventRegistration.php

```diff
@@ -0,0 +1,23 @@
+<?php
+/**
+ * Class for Event Online Registartion BuildForm Hook
+ */
+class CRM_EventsExtras_Hook_BuildForm_EventRegistration extends CRM_EventsExtras_Hook_BuildForm {
+
+  /**
+  *
+  * @param string $$eventTab
+  */
+  public function __construct($eventTab) {
```

Comment:
You don't need to pass this in from the constructor. Each extending class can have a private property that defines what the value of `$eventTab` is.